src/rdb.c

```diff
@@ -2541,7 +2570,19 @@ int rdbLoadRio(rio *rdb, int rdbflags, rdbSaveInfo *rsi) {
          * Similarly if the RDB is the preamble of an AOF file, we want to
          * load all the keys as they are, since the log of operations later
          * assume to work in an exact keyspace state. */
-        if (iAmMaster() &&
+        if (val == NULL) {
+            /* Since we used to have bug that could lead to empty keys
+             * (See #8453), we rather not fail when empty key is encountered
+             * in an RDB file, instead we will silently discard it and
+             * continue loading. */
+            if (error == RDB_LOAD_ERR_EMPTY_KEY) {
+                serverLog(LL_WARNING, "rdbLoadObject failed, detect empty key: %s", key);
```

Comment:
I think it's better to set a counter and maybe report the first occurrence and later the number of total such keys. A log message per key can be devastating for the log file...

Reply:
It bothered me too, but I didn't want to lower the log level. 
Yossi's idea seems best

Reply:
Thanks @oranagra, I'm later.